app/javascript/packs/application.tsx

```diff
@@ -178,15 +178,16 @@ const PerksExternalModalButton = lazy(
 
 const Trophies = lazy(() => import('@/components/track/Trophies'))
 
+import { QueryClient } from '@tanstack/react-query'
 declare global {
   interface Window {
     Turbo: typeof import('@hotwired/turbo/dist/types/core/index')
-    queryCache: QueryCache
+    queryClient: QueryClient
   }
 }
-
-import { QueryCache } from 'react-query'
-window.queryCache = new QueryCache()
+// use query client by pulling it out of the provider with useQueryClient hook
```

Comment:
Is this a TODO? If so, maybe add "TODO: " at the start

Reply:
No, this is how you can access data that is inside the queryClient - same as useContext, called useQueryClient. it is an instruction, or just a note. this is a new behaviour, so I thought i'd make a comment on it.